toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -2073,6 +2079,32 @@ func setGrubCfgSELinux(grubPath string, kernelCommandline configuration.KernelCo
 	return
 }
 
+func setGrubCfgCGroup(grubPath string, kernelCommandline configuration.KernelCommandLine) (err error) {
+	const (
+		cgroupPattern     = "{{.CGroup}}"
+		cgroupv1FlagValue = "systemd.unified_cgroup_hierarchy=0"
+		cgroupv2FlagValue = "systemd.unified_cgroup_hierarchy=1"
+	)
+	var cgroup string
+
+	switch kernelCommandline.CGroup {
+	case configuration.CGroupV2:
+		cgroup = fmt.Sprintf("%s", cgroupv2FlagValue)
+	case configuration.CGroupV1:
+		cgroup = fmt.Sprintf("%s", cgroupv1FlagValue)
+	case configuration.CGroupDefault:
+		cgroup = ""
+	}
+
+	logger.Log.Debugf("Adding CGroupConfiguration('%s') to '%s'", cgroup, grubPath)
+	err = sed(cgroupPattern, cgroup, kernelCommandline.GetSedDelimeter(), grubPath)
+	if err != nil {
```

Comment:
Great work! This is a good opportunity to include some test coverage for this new installutils.go function

Once we get some tests written, coverage can be tracked by running `sudo make go-test-coverage` from your toolkit and downloading the Coverage report html to get confirmation that the new tests cover this new function

Some test recommendations I have are: "TestShouldReturnErrorWhenCGroupPatternIsMissingFromGrubCfg" and "TestShouldSetCorrectCgroupFlagValueForConfiguration"

Reply:
Based on our discussion, marking this resolved until we get feedback from team members